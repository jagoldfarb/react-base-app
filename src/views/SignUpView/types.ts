export interface ISignUpFormData {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  passwordConfirmation: string;
}

export interface ISignUpViewProps {
  onSubmit: (formData: ISignUpFormData) => void;
}
