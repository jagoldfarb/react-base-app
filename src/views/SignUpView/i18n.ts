import i18next from 'i18next';
import { locales } from 'configs/i18n';

i18next.addResources(locales.EN, 'SIGNUP_VIEW', {
  ALREADY_HAVE_ACCOUNT: 'Already have an account?',
  LOGIN: 'Log In',
  SIGNUP: 'Sign Up',
});
