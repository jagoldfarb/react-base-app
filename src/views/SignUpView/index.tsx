import React, { FunctionComponent } from 'react';
import i18next from 'i18next';

import SignUpForm from 'views/SignUpView/components/SignUpForm';

import { ISignUpViewProps } from './types';
import {
  SignUpContainer,
  SignUpContent,
  Title,
  SignUpFormContainer,
  LoginContainer,
  LoginLink,
} from './styles';
import './i18n';

const SignUpView: FunctionComponent<ISignUpViewProps> = (props: ISignUpViewProps) => {
  const { onSubmit } = props;

  return (
    <SignUpContainer data-testid="signup-view">
      <SignUpContent>
        <Title>{i18next.t('SIGNUP_VIEW:SIGNUP')}</Title>
        <SignUpFormContainer>
          <SignUpForm onSubmit={onSubmit} />
        </SignUpFormContainer>
        <LoginContainer>
          {i18next.t('SIGNUP_VIEW:ALREADY_HAVE_ACCOUNT')}
          <LoginLink to="/login">{i18next.t('SIGNUP_VIEW:LOGIN')}</LoginLink>
        </LoginContainer>
      </SignUpContent>
    </SignUpContainer>
  );
};

export default SignUpView;
