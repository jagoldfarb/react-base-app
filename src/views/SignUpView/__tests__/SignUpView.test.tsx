import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import store from 'redux/store';

import SignUpView from '..';

const defaultProps = {
  onSubmit: jest.fn(),
};

describe('SignUp View', () => {
  it('Should render the signup page', () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <SignUpView {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('signup-view')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
