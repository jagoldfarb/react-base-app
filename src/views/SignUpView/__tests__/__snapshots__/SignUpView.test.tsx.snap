// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp View Should render the signup page 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw ghuzIt"
    data-testid="signup-view"
  >
    <div
      class="sc-pFZIQ bfooQu"
    >
      <h1
        class="sc-jrAGrp gANNIP"
      >
        Sign Up
      </h1>
      <div
        class="sc-kEjbxe kdFnRR"
      >
        <form
          class="sc-eCssSg jvctCF"
          data-testid="signup-form"
        >
          <div
            class="sc-bdfBwQ cgTScR"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                First name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="input-First name"
                  maxlength="255"
                  name="firstName"
                  rows="3"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      First name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="sc-bdfBwQ cgTScR"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Last name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="input-Last name"
                  maxlength="255"
                  name="lastName"
                  rows="3"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Last name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="sc-bdfBwQ cgTScR"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="input-Email"
                  maxlength="255"
                  name="email"
                  rows="3"
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="sc-bdfBwQ cgTScR"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  data-testid="input-Password"
                  maxlength="255"
                  name="password"
                  rows="3"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root"
                    data-testid="toggle-password-button"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="sc-bdfBwQ cgTScR"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Password confirmation
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  data-testid="input-Password confirmation"
                  maxlength="255"
                  name="passwordConfirmation"
                  rows="3"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root"
                    data-testid="toggle-password-button"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Password confirmation
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="sc-jSgupP fXngXj"
          >
            By creating an account, you agree and accept our
            <a
              class="sc-gKsewC jHReib"
              href="/terms-and-conditions"
            >
              Terms and Conditions
            </a>
          </div>
          <div
            class="sc-iBPRYJ kKaNZa"
          >
            <button
              class="sc-hKgILt bytOKm"
              data-testid="button"
              type="submit"
            >
              Sign Up
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-iqHYGH loMoME"
      >
        Already have an account?
        <a
          class="sc-crrsfI fpvyaq"
          href="/login"
        >
          Log In
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
